layers/ntp_test.go

```diff
@@ -0,0 +1,244 @@
+//******************************************************************************
+//
+// Copyright 2016 Ross N. Williams. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+//
+//******************************************************************************
+
+package layers
+
+import (
+	"github.com/google/gopacket"
+	"reflect"
+	"testing"
+)
+
+//******************************************************************************
+
+// check_ntp() uses the ntp.go code to analyse the packet bytes as an NTP UDP
+// packet and generate an NTP object. It then compares the generated NTP object
+// with the one provided and raises the alarm if there is any difference.
+// The desc argument is output with any failure message to identify the test.
+//
+func check_ntp(desc string, t *testing.T, packet_bytes []byte, p_expected_ntp *NTP) {
+
+	// Analyse the packet bytes, yielding a new packet object p.
+	//
+	p := gopacket.NewPacket(packet_bytes, LinkTypeEthernet, gopacket.Default)
+	if p.ErrorLayer() != nil {
+		t.Error("Failed to decode packet "+desc+":", p.ErrorLayer().Error())
+	}
+
+	// Ensure that the packet analysis yielded the correct set of layers:
+	//    Link Layer        = Ethernet.
+	//    Network Layer     = IPv4.
+	//    Transport Layer   = UDP.
+	//    Application Layer = NTP.
+	//
+	checkLayers(p, []gopacket.LayerType{
+		LayerTypeEthernet,
+		LayerTypeIPv4,
+		LayerTypeUDP,
+		LayerTypeNTP}, t)
+
+	// Select the Application (NTP) layer.
+	//
+	p_result_ntp, ok := p.ApplicationLayer().(*NTP)
+	if !ok {
+		t.Error("No NTP layer type found in packet in " + desc + ".")
+	}
+
+	// Compare the generated NTP object with the expected NTP object.
+	//
+	if !reflect.DeepEqual(p_result_ntp, p_expected_ntp) {
+		t.Errorf("NTP packet processing failed for packet "+desc+
+			":\ngot  :\n%#v\n\nwant :\n%#v\n\n", p_result_ntp, p_expected_ntp)
+	}
+
+}
+
+//******************************************************************************
+
+func test_01(t *testing.T) {
```

Comment:
Rename these:

```
test_01 -> TestNTPSomeDescription
test_02 -> TestNTPDifferentDescription
test_03 -> TestNTPAnotherDescription
```

Then remove TestNTP... go will handle testing the above individually, which makes it easier to just rerun a single failing test.